python/ray/train/lightning/lightning_trainer.py

```diff
@@ -527,6 +535,13 @@ def _lightning_train_loop_per_worker(config):
     trainer_config["enable_checkpointing"] = True
     model_checkpoint_config["save_last"] = True
 
+    # Convert dirpath to an absolute path under the trial directory
+    ckpt_dirpath = model_checkpoint_config.get("dirpath", None)
+    if ckpt_dirpath and not os.path.isabs(ckpt_dirpath):
+        model_checkpoint_config["dirpath"] = os.path.join(
+            session.get_trial_dir(), ckpt_dirpath
+        )
```

Comment:
One risk:
- Saving to the trial dir will upload the checkpoint again as an artifact. In the cloud checkpointing case.
- I think this is already the default behavior though. We should just make a tracker issue for this problem.

Reply:
Yeah I notice that we already upload the whole trial folder to the cloud storage. Maybe we can setup a flag to determine whether to upload the artifacts or not.

Reply:
There actually is a `sync_artifacts` flag that defaults to True. Can recommend that in the meantime.